# Attempting to Size the Psychiatry Specialty
For context: 

The AAMC https://www.aamc.org/data/workforce/reports/492556/1-1-chart.html lists  892,856 in total active specialty physicans and 38,205 (4.3%) psychiatrists for 2017.

Statista here https://www.statista.com/statistics/209424/us-number-of-active-physicians-by-specialty-area/ lists 525,439 total active "specialty" physicians in March of 2019 of which 54,935 (10.5%) are psychiatrists. 

Keeping in mind there's a shortage of psychiatrists in general: https://news.aamc.org/patient-care/article/addressing-escalating-psychiatrist-shortage/

Here, I'm going to try to scrape individuals from the NPI registry to see how many individuals are listed under the taxonomy of 'psychiatry'



In [63]:
import pandas as pd
import requests
import matplotlib
import numpy as np
import matplotlib.pyplot as plt
import re
import time
from bs4 import BeautifulSoup

In [65]:
# skip=range(0,20000,200)

In [77]:
skip=[1200,2400]

In [74]:
data=[]

In [75]:
#individual test
# link=f'https://npiregistry.cms.hhs.gov/api/?number=&enumeration_type=NPI-2&taxonomy_description=Psychiatry&first_name=&last_name=&organization_name=&address_purpose=&city=&state=&postal_code=&country_code=&limit=200&skip=300&version=2.0'
# response_npi = requests.get(link)
# data_npi = response_npi.json()

In [78]:
for num in skip:
    link=f'https://npiregistry.cms.hhs.gov/api/?number=&enumeration_type=NPI-2&taxonomy_description=Psychiatry&first_name=&last_name=&organization_name=&address_purpose=&city=&state=&postal_code=&country_code=&limit=200&skip={num}&version=2.1'
    response_npi = requests.get(link)
    data_npi = response_npi.json()
    results=data_npi['results']
    print('---------------------------------')
    print(results[0]['basic']['authorized_official_first_name'])
    print(len(results))
#     time.sleep(1)
    for result in results[:len(results)-1]:
        person=result['basic']
        print(result['basic']['authorized_official_first_name'])
        for address in result['addresses']:
            column_name='zip_'+ str(result['addresses'].index(address)+1)
            person[column_name]=address['postal_code']
#             data.append(person)

---------------------------------
CHUCK
200
CHUCK
DAVID
MARK
DAVID
ALBERTA
HATTI
MICKEY
NANCY
DARLENE
NEIL
MANUEL
MANOJ
RALPH
PAUL
SWAMPNA
GAIL
ANDREW
JUDE
JACQUELINE
BRONSON
LISA
MONICA
NORMAN
TROY
TALIN
MELISSA
STEVEN
BRYON
RANDOLPH
DEBORAH
JENNIFER
PENELOPE
BENNETT
FREDERICK
FREDERICK
FREDERICK
FREDERICK
MICHAEL
KEITH
MICHAEL
BENNETT
RICHARD
DANIEL
LEISA
STEVE
HITEN
ANTHONY
LUCKOSE
THOMAS
LAWRENCE
REBECCA
ANGELA
KAREN
FRANCES
EDWIN
LINDA
WILLIAM
HOWARD
FRANCISCO
PATRICK
FREDERICK
CRYSTAL
RICK
MAXWELL
D
D
D
ROBERT
KENNETH
DAVID
REBECCA
EDWARD
AMY
CHARLES
JOSEPH
ERIC
ROGER
LEFAWN
CHERYL
GREG
DAVIS
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
DAVE
WALTER
MICHAEL
CARL
LOUISE
DENNIS
JASON
OMAR
JORRI
MICHAEL
JEFFREY
WILLIAM
JILL
MELISSA
ANTHONY
JEROME
CRAIG
SUSAN
JASMINE
DARREL
CANDACE
ABDUL
ALEXANDRA
STEPHEN
ALBERTO
LARRY
ROYAL
FEBE
GEOFFREY
RUSSELL
RUTH
MONIQUE
SYAM
JOSH
CHRIS
IAN
DEBORAH
ROBIN
FREDERICK
NITIN
ANN MARE
TIMOTHY
R

In [70]:
df=pd.DataFrame(data)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39800 entries, 0 to 39799
Data columns (total 20 columns):
authorized_official_credential             19834 non-null object
authorized_official_first_name             39800 non-null object
authorized_official_last_name              39800 non-null object
authorized_official_middle_name            19854 non-null object
authorized_official_name_prefix            18482 non-null object
authorized_official_name_suffix            1016 non-null object
authorized_official_telephone_number       39800 non-null object
authorized_official_title_or_position      39800 non-null object
deactivation_date                          2106 non-null object
enumeration_date                           39800 non-null object
last_updated                               39800 non-null object
name                                       39800 non-null object
organization_name                          39800 non-null object
organizational_subpart                     39800

In [71]:
df.duplicated().value_counts()

True     38606
False     1194
dtype: int64

In [79]:
df=df.drop_duplicates(keep='first')

In [81]:
df.shape

(1194, 20)

So, turns out there are only 1,194 Psychiatrists and Neurologists listed which is an estimated: 2-3% of the workforce. 